File: frontend/src/types/note.ts
export interface NoteTreeNode {
  id: string;
  name: string;
  type: "folder" | "file";
  google_file_id: string;
  children?: NoteTreeNode[];
}

export interface NoteTreeResponse {
  folder_id: string;
  tree: NoteTreeNode[];
}

export interface Note {
  id: number;
  user_id: number;
  google_file_id: string | null;
  title: string;
  folder_path: string | null;
  mime_type: string | null;
  last_synced_at: string;
  created_at: string;
  updated_at: string;
}

export interface LinkedNoteBrief {
  id: number;
  title: string;
  folder_path: string | null;
  google_file_id: string | null;
  file_id?: string;
}

File: backend/app/schemas/note.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class NoteResponse(BaseModel):
    id: int
    user_id: int
    google_file_id: str
    title: str
    folder_path: Optional[str] = None
    mime_type: Optional[str] = None
    last_synced_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class LinkedNoteBrief(BaseModel):
    id: int
    title: str
    folder_path: Optional[str] = None
    google_file_id: str

    model_config = {"from_attributes": True}


class LinkedJobBrief(BaseModel):
    id: int
    title: str
    company: Optional[str] = None
    status: str

    model_config = {"from_attributes": True}


class NoteCreate(BaseModel):
    title: str
    content: str


class NoteTreeNode(BaseModel):
    id: Optional[str] = None
    name: str
    type: str  # "folder" or "file"
    google_file_id: str
    children: list[NoteTreeNode] = []


class NoteTreeResponse(BaseModel):
    folder_id: str
    tree: list[NoteTreeNode]
